core/trino-main/src/main/java/io/trino/operator/Driver.java

```diff
@@ -385,7 +385,7 @@ private ListenableFuture<Void> processInternal(OperationTimer operationTimer)
 
         processNewSources();
 
-        // If there is only one operator, finish it
+        // One of the operators is already finished (and removed from activeOperators list). Finish remaining operators.
```

Comment:
It doesn't finish all remaining operators, but only the topmost one. Otherwise LGTM.